File: 0199-二叉树的右视图/Solution.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Stack;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if(root == null){ return res; }
        LinkedList<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while(!queue.isEmpty()){
            int num = queue.size();
            if(num == 0){ return res;}
            res.add(queue.peekLast().val);
            for (int i = 0; i < num; i++) {
                TreeNode curr = queue.poll();
                if(curr.left != null){ queue.offer(curr.left); }
                if(curr.right != null){ queue.offer(curr.right); }
            }
        }
        return res;
    }
}

File: 0199-二叉树的右视图/solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        ret = []
        if not root:
            return ret
        queue = [root]
        while queue:
            ret.append(queue[-1].val)
            for _ in range(len(queue)):
                tmp = queue.pop(0)
                if tmp.left:
                    queue.append(tmp.left)
                if tmp.right:
                    queue.append(tmp.right)
        return ret